service/controller/v19/resource/ipam/create.go

```diff
@@ -34,57 +33,82 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	r.logger.LogCtx(ctx, "level", "debug", "message", "finding out if subnet needs to be allocated for cluster")
 
 	if key.ClusterNetworkCIDR(customObject) == "" {
-		var subnetCIDR net.IPNet
-		// TODO(tuommaki): Remove this when all tenant clusters are upgraded to this
-		// version and have subnet allocation in their Status field.
+		// TODO remove the status checks for older clusters when all tenant clusters
+		// are upgraded to this version and have subnet allocation in their Status
+		// field.
 		//
 		//     https://github.com/giantswarm/giantswarm/issues/4192
 		//
-		if key.CIDR(customObject) != "" && strings.ToLower(key.CIDR(customObject)) != "deprecated" {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "found out allocated cluster CIDR from legacy field in CR")
+		var statusAZs []v1alpha1.AWSConfigStatusAWSAvailabilityZone
+		var subnetCIDR net.IPNet
+		if customObject.ClusterStatus().HasUpdatedCondition() {
```

Comment:
```suggestion
		if customObject.ClusterStatus().HasUpdatingCondition() {
```

This must be executed in the very beginning of upgrade and at that point the status is `Updating`, right?

Reply:
The idea here is that we have two types of situations to manage. A newly created cluster and older clusters. Older clusters have the `Updated` status condition. So this is already correct.

Reply:
My understanding is cluster with `Updated` is a legacy cluster so we have to migrate it.

Reply:
What is missing for the migration? The code within this conditional scope is the migration code.

Reply:
Uhm, if I spin cluster with `v18` and upgrade it to `v19`, it gets `Updated` status only once the whole cluster is rolled with new version. Hence, this would behave incorrectly here: It would treat that as a new cluster and generate a new subnet and select random AZ when it actually should pick the existing values from legacy fields.

Reply:
Oh that is true. Then it should be either `Updating` or `Updated` which we consider here. That should be good enough right? Otherwise I am not sure which indicator we should use here.

Reply:
Sorted also considering the `Updating` status condition.